molo/surveys/models.py

```diff
@@ -300,9 +301,10 @@ def serve_questions(self, request):
         """
         context = self.get_context(request)
         # this will only return a page if there is a translation
-        page = context['page'].get_translation_for(
-            locale=request.LANGUAGE_CODE, site=request.site)
-        if page:
+        page = get_translation_for(
+            [context['page']],
+            locale=request.LANGUAGE_CODE, site=request.site)[0]
+        if not page.language.is_main_language:
```

Comment:
Will we always be on the main language page when we are here? or will this method still be called if a user is looking at the french translation for a survey and changes to the main language (or a 3rd language)?
If the latter then we should just check that the page returned by the translation call isn't the same page we're on.

Reply:
if it was the same page we are on then get_translation_for would return that page

Reply:
so this should be the same behaviour

Reply:
I was dealing with infinite redirects here so I handled this by first checking if the page that was returned is the page that we're already on `if page != context['page']:` rather than checking for the main language. I'm suspicious that your fix might not handle the infinite loop issue